### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by kumarabd**
Signed-off-by: kumarabd <abishekkumar92@gmail.com>

There needs to be some modifications done from the proto file too. @leecalcote

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,47 +1,59 @@
...
 
-[![Docker Pulls](https://img.shields.io/docker/pulls/layer5/meshery.svg)](https://hub.docker.com/r/layer5/meshery)
-[![Go Report Card](https://goreportcard.com/badge/github.com/layer5io/meshery)](https://goreportcard.com/report/github.com/layer5io/meshery)
-[![Build Status](https://github.com/layer5io/meshery/workflows/Meshery/badge.svg)](https://github.com/layer5io/meshery/actions)
-[![GitHub](https://img.shields.io/github/license/layer5io/meshery.svg)](LICENSE)
-[![GitHub issues by-label](https://img.shields.io/github/issues/layer5io/meshery/help%20wanted.svg)](https://github.com/layer5io/meshery/issues?q=is%3Aopen+is%3Aissue+label%3A"help+wanted")
-[![Website](https://img.shields.io/website/https/layer5.io/meshery.svg)](https://layer5.io/meshery/)
-[![Twitter Follow](https://img.shields.io/twitter/follow/layer5.svg?label=Follow&style=social)](https://twitter.com/intent/follow?screen_name=mesheryio)
+# Meshery Adapter for Kuma
```
Comment (leecalcote) [inline README.md:3]: Inclusion of Kuma specifics here might be accidental.

Comment (kumarabd) [reply README.md]: @leecalcote  yes accidental, fixed now.

File: adaptor/adaptor.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package adaptor
+
+import (
+	"github.com/kumarabd/gokit/logger"
```
Comment (leecalcote) [inline adaptor/adaptor.go:4]: Perhaps this should be https://github.com/go-kit/kit?

File: adaptor/error.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package adaptor
+
+import (
+	"fmt"
+
+	"github.com/kumarabd/gokit/errors"
```
Comment (leecalcote) [inline adaptor/error.go:6]: A reference to update pending https://github.com/go-kit/kit vs https://github.com/layer5io/meshkit

File: adaptor/spec.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	Version string `json:"version"`
+}
+
+// GetName returns the name of the mesh
+func (h *handler) GetName() string {
+	spec := &Spec{}
+	err := h.config.MeshSpec(&spec)
+	if err != nil {
+		h.log.Err("1000", err.Error())
+		return "Not set"
```
Comment (leecalcote) [inline adaptor/spec.go:16]: "Missing service mesh name" - is this more descriptive?

File: api/grpc/error.go [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package grpc
+
+import (
+	"fmt"
+
+	"github.com/kumarabd/gokit/errors"
```
Comment (leecalcote) [inline api/grpc/error.go:6]: A reference to track.

File: error_codes.txt [hunk 6]
```diff
@@ -0,0 +1,15 @@
+Error codes
```
Comment (leecalcote) [inline error_codes.txt:1]: To be centrally defined in https://docs.meshery.io later.

File: internal/config/defaults.go [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+	"os/user"
+)
+
+var (
+
+	// server holds server configuration
+	server = map[string]string{
+		"name":    "<adaptor-name>-adaptor",
+		"port":    "10007",
+		"version": "v1.0.0",
```
Comment (leecalcote) [inline internal/config/defaults.go:14]: I wonder if we might start with "v0.1.0"

File: internal/config/defaults.go [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+	"fmt"
+	"os/user"
+)
+
+var (
+
+	// server holds server configuration
+	server = map[string]string{
+		"name":    "<adaptor-name>-adaptor",
+		"port":    "10007",
```
Comment (leecalcote) [inline internal/config/defaults.go:13]: I wonder if we might stub out the port number here.

Comment (leecalcote) [inline internal/config/defaults.go:13]: I wonder if we might stub out the port number here.

File: internal/config/error.go [hunk 9]
```diff
@@ -0,0 +1,17 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/kumarabd/gokit/errors"
```
Comment (leecalcote) [inline internal/config/error.go:6]: Reference to track

File: internal/config/local.go [hunk 10]
```diff
@@ -0,0 +1,65 @@
+package config
+
+import (
+	"github.com/kumarabd/gokit/utils"
```
Comment (leecalcote) [inline internal/config/local.go:4]: Reference to track

File: internal/config/local.go [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+// GetKey gets a key value from local store
+func (l *Local) GetKey(key string) string {
+	return l.store[key]
+}
+
+// Server provides server specific configuration
+func (l *Local) Server(result interface{}) error {
+	d := `{
+		"name":    "<adaptor-name>-adaptor",
+		"port":    "10007",
```
Comment (leecalcote) [inline internal/config/local.go:35]: Port number to stub out

File: internal/config/local.go [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+		"traceurl": "http://localhost:14268/api/traces",
+		"version": "v1.0.0"
```
Comment (leecalcote) [inline internal/config/local.go:37]: Shift to minor version to "init" an adapter.

File: main.go [hunk 13]
```diff
@@ -1,63 +1,58 @@
...
+	"time"
 
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/grpclog"
-
-	"github.com/sirupsen/logrus"
-
-	"github.com/layer5io/meshery-<adapter>/<adapter>"
-	mesh "github.com/layer5io/meshery-<adapter>/meshes"
+	"github.com/kumarabd/gokit/logger"
```
Comment (leecalcote) [inline main.go:8]: reference

File: main.go [hunk 14]
```diff
@@ -1,63 +1,58 @@
...
 
-func init() {
-	log = grpclog.NewLoggerV2(os.Stdout, os.Stdout, os.Stdout)
-	grpclog.SetLoggerV2(log)
-}
+	// Initialize application specific configs and dependencies
+	// App and request config
+	cfg, err := config.New(configProvider)
+	if err != nil {
+		fmt.Println("Config Init Failed", err.Error())
```
Comment (leecalcote) [inline main.go:26]: `println` or `log`?

File: main.go [hunk 15]
```diff
@@ -1,63 +1,58 @@
...
+	handler := adaptor.New(cfg, log)
+	service.Handler = handler
+	service.StartedAt = time.Now()
+
+	// Server Initialization
+	log.Info("Adaptor Started")
+	err = grpc.Start(service, traceProvider)
 	if err != nil {
-		logrus.Fatalln("Failed to listen:", err)
+		log.Err("Adaptor crashed!!", err.Error())
```
Comment (leecalcote) [inline main.go:55]: Remove "!!". Provide error code number?

**Comment by leecalcote** [comment]
Reviewing old PR‘s…

**Pull request merged**